### DISCUSSION TITLE
How to generate secrets-token.json file when enabled event data persistence?

### DISCUSSION THREAD

**Discussion by zenosp** [edgexfoundry/community#21] [Q&A]
Hi guys,

when I enabled StoreAndForward, the app-service-configurable start failed.

I started the edgex in native mode on RISC-V ARCH CPU, Only a few microservices are used such as core-data, metadata, core-command, device-service, app-service, and redis-service. I konw nothing about edgex security. 

I want to know how to make it work with a few commands, It is better to have detailed operation instructions.

thanks. 

`$ ./app-service-configurable               
level=INFO ts=2022-10-21T10:10:21.559681577Z app=app-<profile> source=service.go:466 msg="Starting app- 0.0.0 "
level=INFO ts=2022-10-21T10:10:21.560444199Z app=app- source=config.go:373 msg="Loaded service configuration from ./res/configuration.toml"
level=INFO ts=2022-10-21T10:10:21.561438065Z app=app- source=secret.go:52 msg="Creating SecretClient"
level=INFO ts=2022-10-21T10:10:21.561494511Z app=app- source=secret.go:59 msg="Reading secret store configuration and authentication token"
level=WARN ts=2022-10-21T10:10:21.56173251Z app=app- source=secret.go:96 msg="Retryable failure while creating SecretClient: open /tmp/edgex/secrets/app-mqtt-export/secrets-token.json: no such file or directory"
level=INFO ts=2022-10-21T10:10:22.562457183Z app=app- source=secret.go:59 msg="Reading secret store configuration and authentication token"
level=WARN ts=2022-10-21T10:10:22.562637208Z app=app- source=secret.go:96 msg="Retryable failure while creating SecretClient: open /tmp/edgex/secrets/app-mqtt-export/secrets-token.json: no such file or directory"`

**Comment by lenny-goodell** [comment]
You need to set environment variable `EDGEX_SECURITY_SECRET_STORE=false`
